polynote-frontend/polynote/ui/component/notebook/notebook.ts

```diff
@@ -51,6 +66,10 @@ export class Notebook extends Disposable {
         const cellsHandler = notebookState.cellsHandler
 
         const handleAddedCells = (added: Partial<Record<number, CellState>>, cellOrderUpdate: UpdateResult<number[]>) => {
+
+            // if no cells exist yet, we'll need to do an initial layout after adding the cells.
+            const needLayout = Object.keys(this.cells).length === 0;
```

Comment:
I'm not super pleased about this logic... it's basically saying "the first time we add all the cells, we'll do a layout afterward." The problem is that when the `notebook.loaded` promise resolves, the cells haven't been added to the DOM yet, so it's too early to do layout at that point.

Reply:
> The problem is that when the notebook.loaded promise resolves, the cells haven't been added to the DOM yet, so it's too early to do layout at that point.

I ran into this problem as well. There's a bug in the way that that the notebook loading promise is set up. Just pushed my fix as a PR here: https://github.com/polynote/polynote/pull/1112